google-cloud-bigquery/src/main/java/com/google/cloud/bigquery/TableDefinition.java

```diff
@@ -145,7 +146,7 @@ Table toPb() {
   }
 
   @SuppressWarnings("unchecked")
-  static <T extends TableDefinition> T fromPb(Table tablePb) {
+  static <T extends TableDefinition> T fromPb(Table tablePb) throws IOException, Throwable {
```

Comment:
Ditto. I'm concerned that this might be a breaking change. I guess it's okay since it's protected?

Let's be more specific than "Throwable" and use IllegalArgumentException, NullPointerException.